### DISCUSSION TITLE
ARMv7 support

### DISCUSSION THREAD

**Discussion by hez** [tailwindlabs/tailwindcss#7403] [Ideas]
A lot of Raspberry Pi deployments use the ARMv7 Linux arch, it would be really nice if this could build and create releases for.

**Comment by hez** [comment]
After looking at https://github.com/tailwindlabs/tailwindcss/pull/6693/files I went a head and created the following branch https://github.com/hez/tailwindcss/tree/feature/armv7 but couldn't figure out how to get it running. Open to suggestions here.

**Comment by hez** [comment]
I ran npm run build in the standalone directory and got some pretty good output. Though I am not sure about the error `Warning Cannot find module 'tailwindcss/lib/cli' from '/Users/hez/Work/tailwindcss/standalone-cli'  in /Users/hez/Work/tailwindcss/standalone-cli/standalone.js`

```
standalone-cli (feature/armv7)> npm run build

> tailwindcss-standalone@0.0.0 prebuild
> rimraf dist


> tailwindcss-standalone@0.0.0 build
> pkg standalone.js --out-path dist --targets node16-macos-x64,node16-macos-arm64,node16-win-x64,node16-linuxstatic-x64,node16-linuxstatic-arm64,node16-linuxstatic-armv7 --compress Brotli --no-bytecode --public-packages "*" --public

> pkg@5.3.3
compression:  Brotli
> Warning Cannot find module 'tailwindcss/lib/cli' from '/Users/hez/Work/tailwindcss/standalone-cli'  in /Users/hez/Work/tailwindcss/standalone-cli/standalone.js


> tailwindcss-standalone@0.0.0 postbuild
> move-file dist/standalone-macos-x64 dist/tailwindcss-macos-x64 && move-file dist/standalone-macos-arm64 dist/tailwindcss-macos-arm64 && move-file dist/standalone-win-x64.exe dist/tailwindcss-windows-x64.exe && move-file dist/standalone-linuxstatic-x64 dist/tailwindcss-linux-x64 && move-file dist/standalone-linuxstatic-arm64 dist/tailwindcss-linux-arm64 && move-file dist/standalone-linuxstatic-armv7 dist/tailwindcss-linux-armv7
```

```
standalone-cli (feature/armv7)> ls -la dist/
total 225728
drwxr-xr-x  8 hez staff      256 Feb  9 20:41 ./
drwxr-xr-x 10 hez staff      320 Feb  9 20:40 ../
-rwxr-xr-x  1 hez staff 34880154 Feb  9 20:40 tailwindcss-linux-arm64*
-rwxr-xr-x  1 hez staff 32328414 Feb  9 20:41 tailwindcss-linux-armv7*
-rwxr-xr-x  1 hez staff 37993786 Feb  9 20:40 tailwindcss-linux-x64*
-rwxr-xr-x  1 hez staff 42389600 Feb  9 20:40 tailwindcss-macos-arm64*
-rwxr-xr-x  1 hez staff 47331888 Feb  9 20:40 tailwindcss-macos-x64*
-rw-r--r--  1 hez staff 34469865 Feb  9 20:40 tailwindcss-windows-x64.exe
```

**Comment by RomanKotov** [comment]
It seems that building standalone cli requires some extra steps, according to the GitHub Actions (https://github.com/tailwindlabs/tailwindcss/blob/2882d4c9eeed91f1e7b48b40c0e57ea29b2666d9/.github/workflows/build-cli.yml)

So, the whole workflow now is like:
- go to tailwind project root
- install dependencies from the project root `npm install`
- build Tailwindcss `npm run prepublishOnly`
- navigate to standalone-cli folder `cd standalone-cli`
- install dependencies there `npm install`
- build a standalone cli `npm run build`

I have create a PR for this feature https://github.com/tailwindlabs/tailwindcss/pull/9084

**Comment by spicychickensauce** [comment]
Unfortunately the armv7 support was dropped with Tailwind 4.
Could we get it back? Any reason why it was dropped with v4?
